src/hotspot/cpu/riscv/vm_version_riscv.cpp

```diff
@@ -289,17 +293,9 @@ void VM_Version::c2_initialize() {
 
   if (!UseRVV) {
     FLAG_SET_DEFAULT(SpecialEncodeISOArray, false);
```

Comment:
Could you please remove this line while you are on it? I don't think we need to reset `SpecialEncodeISOArray` here.

Reply:
I'm not sure, as there are some code in share/classfile/vmIntrinsics.cpp which depends on SpecialEncodeISOArray's value:
```
  case vmIntrinsics::_encodeISOArray:
  case vmIntrinsics::_encodeAsciiArray:
  case vmIntrinsics::_encodeByteISOArray:
    if (!SpecialEncodeISOArray) return true;
    break;
```

Reply:
The reason is that `SpecialEncodeISOArray` is a DIAGNOSTIC option which I think is for debugging / trouble-shooting purpose. The other four ones are `SpecialStringCompareTo`, `SpecialStringIndexOf`, `SpecialStringEquals` and `SpecialArraysEquals`. This code snippet checking for `SpecialEncodeISOArray` is in function `vmIntrinsics::disabled_by_jvm_flags`. So I think we are safe to have this change. In fact, `SpecialEncodeISOArray` and the others are not checked or modified by other platforms either.

Reply:
It makes sense, Thanks!